tests/unit/borg/test_create.py

```diff
@@ -582,7 +582,52 @@ def test_create_archive_with_patterns_calls_borg_with_patterns_including_convert
         },
         storage_config={},
         local_borg_version='1.2.3',
-        global_arguments=flexmock(log_json=False),
+        global_arguments=flexmock(log_json=False, used_config_paths=[]),
+    )
+
+
+def test_create_archive_with_sources_and_used_config_paths_calls_borg_with_sources_and_config_paths():
+    flexmock(module.borgmatic.logger).should_receive('add_custom_log_levels')
+    flexmock(module.logging).ANSWER = module.borgmatic.logger.ANSWER
+    flexmock(module).should_receive('collect_borgmatic_source_directories').and_return([])
+    flexmock(module).should_receive('deduplicate_directories').and_return(
+        ('foo', 'bar', '/etc/borgmatic/config.yaml')
+    )
+    flexmock(module).should_receive('map_directories_to_devices').and_return({})
+    flexmock(module).should_receive('expand_directories').and_return(())
+    flexmock(module).should_receive('pattern_root_directories').and_return([])
+    flexmock(module.os.path).should_receive('expanduser').and_raise(TypeError)
+    flexmock(module).should_receive('expand_home_directories').and_return(())
+    flexmock(module).should_receive('write_pattern_file').and_return(None)
+    flexmock(module).should_receive('make_list_filter_flags').and_return('FOO')
+    flexmock(module.feature).should_receive('available').and_return(True)
+    flexmock(module).should_receive('ensure_files_readable')
+    flexmock(module).should_receive('make_pattern_flags').and_return(())
+    flexmock(module).should_receive('make_exclude_flags').and_return(())
+    flexmock(module.flags).should_receive('make_repository_archive_flags').and_return(
+        (f'repo::{DEFAULT_ARCHIVE_NAME}',)
+    )
+    environment = {'BORG_THINGY': 'YUP'}
+    flexmock(module.environment).should_receive('make_environment').and_return(environment)
+    flexmock(module).should_receive('execute_command').with_args(
+        ('borg', 'create') + REPO_ARCHIVE_WITH_PATHS + ('/etc/borgmatic/config.yaml',),
+        output_log_level=logging.INFO,
+        output_file=None,
+        borg_local_path='borg',
+        working_directory=None,
+        extra_environment=environment,
+    )
+
+    module.create_archive(
+        dry_run=False,
+        repository_path='repo',
+        location_config={
+            'source_directories': ['foo', 'bar'],
+            'repositories': ['repo'],
+        },
+        storage_config={},
+        local_borg_version='1.2.3',
+        global_arguments=flexmock(log_json=False, used_config_paths=['/etc/borgmatic/config.yaml']),
```

Comment:
So I did this - 

```python
def test_create_archive_with_sources_and_used_config_paths_calls_borg_with_sources_and_config_paths():
    flexmock(module.borgmatic.logger).should_receive('add_custom_log_levels')
    flexmock(module.logging).ANSWER = module.borgmatic.logger.ANSWER
    flexmock(module).should_receive('collect_borgmatic_source_directories').and_return(['/etc/borgmatic/config.yaml'])
    flexmock(module).should_receive('deduplicate_directories').and_return(
        ('foo', 'bar', '/etc/borgmatic/config.yaml')
    )
    flexmock(module).should_receive('map_directories_to_devices').and_return({})
    flexmock(module).should_receive('expand_directories').with_args(
        ('foo', 'bar', '/etc/borgmatic/config.yaml')
    ).and_return(())
```

But got this error:

"Arguments for call expand_directories did not match expectations:\n  Received call:\texpand_directories(['/etc/borgmatic/config.yaml'])\n  Expected call[1]:\texpand_directories(directories=('foo', 'bar', '/etc/borgmatic/config.yaml'))"

Why didn't expand_directories recieve the location_config as args? And this is what I should be testing right?

Reply:
I'm guessing because `expand_directories()` is called twice—once inside of `map_directories_to_devices()` and once above, on line 346. And I bet that first call is the one where the expectation isn't matched...

Reply:
> And this is what I should be testing right?

And yeah, I think this is probably the right thing to test given the constraints. It's obnoxious to have to do so much mocking, but I don't see a better way to do it short of switching to an integration test.